### PR TITLE
Remove reference not found errors from `jsx: preserve`

### LINKED ISSUE #60603: Since 5.7.2: Using JSX fragments requires fragment factory 'React' to be in scope, but it could not be found.
### 🔎 Search Terms

Search/Keywords: Using JSX fragments requires fragment factory 'React' to be in scope, but it could not be found.

### 🕗 Version & Regression Information

Since 5.7.2. No 5.7.0 or 5.7.1 available. OK in 5.6.3

### ⏯ Playground Link

[Best effort Playground](https://www.typescriptlang.org/play/?ssl=13&ssc=60&pln=11&pc=1#code/JYWwDg9gTgLgBAbzgZVGANgUwMIXBAO0wPgF84AzKPOAchgFcCBzAZxmNrgHpu4YAnmEysAXHAAMAOgCMUgKwyAsAChVmAB6RYcAMboAhq1ZwAMhGbNMAEwDyDeJo4FrJ1OCy58REolVwAvUJneABeOAAKAEo4UIA+SP9A5IAeBN44AEk4VkxMfgALfOQAUWxbADkAEThMKGooOCKofOACOBABWoIAN2BqAhBiGCTksbgUgpk480sbOHsYFO4p9L5sqsraABUcvMLgEzqGuDaOruI+gaGSUbHluLuo1VJVVQzdAoMWNuY4beQcB6dVYwEI-AgQIUUgAbFIAMx7TAgEwwSGfb5WQr5ADu0Gsv3efAgFGxlH67Fq9WgkQIEHg31OxgYrXanW6V0INxgUQANHBvtYBehWBAiXAWiAIMDUUU4AAxKAGZjcqkNflGU7wawQESnNkXXr9LnDVRAA)

### 💻 Code

```ts
    <>
        <h1>Logged Out</h1>
    </>
```

### 🙁 Actual behavior

Since TS 5.7.2 JSX Fragments are no longer accepted.

Compile error:  error TS2879: Using JSX fragments requires fragment factory 'React' to be in scope, but it could not be found.

### 🙂 Expected behavior

JSX Fragments should continue to work

### Additional information about the issue

I have a custom JSX setup, I'm not using React. `babel-preset-react` is used to transform JSX. This grabs `Fragment` from the JSX runtime provided and passes this as the node type when creating the element via the JSX runtime. It has been working since TS 4 until now.

Compiled output:
```ts
import { jsx as _jsx, Fragment as _Fragment } from "tungsten/jsxRuntime.js";
...
    _jsx(_Fragment, {
        children: _jsx("h1", {
            children: "Logged Out"
        })
    })
```

The error message about React is odd, JSX and React are independent, why would React anything be needed to use JSX fragments? `babel-preset-react` with `runtime: "automatic"` loads jsx functions and the `Fragment` constant from a specified JSX runtime and does not rely on anything called `React`.

### PR DESCRIPTION
**Pull request by iisaduan**
Fixes #60603

Cleans up a lot of errors that shouldn't have been included with #59933. From discussion with @andrewbranch:

- `preserve` shouldn't perform the check
- `react` should check for `React` in scope, and `jsx/jsxFrag` pragmas should check if entity name is in scope
- `react-jsx`, `react-jsx-dev`, settings and `jsxImportSource` pragma should check that the corresponding module can be imported

### CODE REVIEW

File: tests/baselines/reference/jsxFragmentFactoryReference(jsx=react-jsx).errors.txt [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+
+==== jsxFragmentFactoryReference.tsx (2 errors) ====
+    export class LoggedOut {
+        content = () => (
+            <>        </>
+            ~~
+!!! error TS2792: Cannot find module 'react/jsx-runtime'. Did you mean to set the 'moduleResolution' option to 'nodenext', or to add aliases to the 'paths' option?
+            ~~
+!!! error TS2879: Using JSX fragments requires fragment factory 'React' to be in scope, but it could not be found.
```
Comment (weswigham) [inline tests/baselines/reference/jsxFragmentFactoryReference(jsx=react-jsx).errors.txt:12]: Isn't this just... not true? Under `react-jsx`, `Fragment` is imported from the jsx runtime module (automatically) - `React` shouldn't be needed in scope at all, afaik.

**Comment by iisaduan** [comment]
I found that a lot of the errors generated in the original PR seem to be incorrect as well. The most recent change removes those, and adds some more tests to show the behavior with the rest of the `jsx` options

**Comment by newives** [comment]
Is there any progress?

**Review by andrewbranch — APPROVED** [review]

**Comment by iisaduan** [comment]
@typescript-bot cherry-pick this to release-5.7

**Pull request merged**